adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -0,0 +1,477 @@
+                    self._noun_for_object(first_object)
+                # multiple objects of different types
+                else:
+                    for object_ in self.situation.salient_objects:
+                        print(object_)
+                        if not self._only_translate_if_referenced(object_):
+                            self._noun_for_object(object_)
+
+            # TODO: handle persisting relations
+            for persisting_relation in self.situation.always_relations:
+                self._translate_relation(action, persisting_relation)
+
+            return immutableset(
+                [
+                    self.generator._dependency_tree_linearizer.linearize(
+                        DependencyTree(self.dependency_graph)
+                    )
+                ]
+            )
+
+        """Get the noun for the object in a given situation"""
+
+        def _noun_for_object(
+            self,
+            _object: SituationObject,
+            *,
+            syntactic_role_if_known: Optional[DependencyRole] = None,
+        ) -> DependencyTreeToken:
+
+            # if we already have a mapping for a noun, we're done
+            if _object in self.objects_to_dependency_nodes:
+                return self.objects_to_dependency_nodes[_object]
+
+            # the number of this object that is in the scene
+            count = self.object_counts[_object.ontology_node]
+
+            # make sure there is a corresponding ontology node
+            if not _object.ontology_node:
+                raise RuntimeError(
+                    f"Don't know how to handle objects which don't correspond to "
+                    f"an ontology node currently: {_object}"
+                )
+
+            # check if the situation object is the speaker
+            if IS_SPEAKER in _object.properties:
+                noun_lexicon_entry = ME
+
+            # check if the situation object is the addressee
+            elif IS_ADDRESSEE in _object.properties:
+                noun_lexicon_entry = YOU
+
+            # if not wo or ni, then just an object
+            else:
+                noun_lexicon_entry = self._unique_lexicon_entry(
+                    _object.ontology_node  # pylint:disable=protected-access
+                )
+
+            # create a dependency node for the noun
+            dependency_node = DependencyTreeToken(
+                noun_lexicon_entry.base_form,
+                noun_lexicon_entry.part_of_speech,
+                morphosyntactic_properties=noun_lexicon_entry.intrinsic_morphosyntactic_properties,
+            )
+            self.dependency_graph.add_node(dependency_node)
+
+            # add a classifier if necessary
+            self.add_classifier(
+                _object, count, dependency_node, noun_lexicon_entry=noun_lexicon_entry
+            )
+
+            # TODO: handle X_IS_Y; not implemented in syntax yet either
```

Comment:
Is this just not implemented for Chinese, or not implemented for English as well?